x-pack/plugins/cross_cluster_replication/public/app/constants/form_schemas.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import Joi from 'joi';
+import { INDEX_ILLEGAL_CHARACTERS_VISIBLE } from 'ui/indices';
+
+import { indexNameValidator } from '../services/input_validation';
+
+const indexNameIllegalCharacters = INDEX_ILLEGAL_CHARACTERS_VISIBLE.join(' ');
+
+/* eslint-disable max-len */
+export const followerIndexFormSchema = {
```

Comment:
I don't think this needs to be a blocker, but I wonder about the utility of extracting this data structure. With the exception of the advanced settings, we don't traverse it and each node is referred to once within `follower_index_form.js`. I'd suggest moving this data into `follower_index_form.js` to localize it with the other rendering logic, which will make it easier to see what's being rendered. Then you won't need the wrapper object and we could just have individual `nameSchema`, `leaderIndexSchema`, and `advancedSchema` vars. And I think `advancedSchema` should be an array to ensure the order of the fields is deterministic. Anyway just some thoughts, not a blocker.

Reply:
++ on array for ordering